src/hooks/useRoomMembers.ts

```diff
@@ -43,42 +42,28 @@ type RoomMemberCountOpts = {
      * Wait time between room member count update
      */
     throttleWait?: number;
-    /**
-     * Whether to include functional members (bots, etc...) in the room count
-     * @default true
-     */
-    includeFunctional: boolean;
 };
 
 // Hook to simplify watching Matrix Room joined member count
-export const useRoomMemberCount = (
-    room: Room,
-    opts: RoomMemberCountOpts = { throttleWait: 250, includeFunctional: true },
-): number => {
+export const useRoomMemberCount = (room: Room, opts: RoomMemberCountOpts = { throttleWait: 250 }): number => {
     const [count, setCount] = useState<number>(room.getJoinedMemberCount());
 
-    const { throttleWait, includeFunctional } = opts;
+    const { throttleWait } = opts;
+
+    const onMembershipUpdate = useCallback(() => {
+        setCount(room.getJoinedMemberCount());
+    }, [room]);
 
     useTypedEventEmitter(
         room.currentState,
         RoomStateEvent.Members,
-        throttle(
-            () => {
-                // At the time where `RoomStateEvent.Members` is emitted the
-                // summary API has not had a chance to update the `summaryJoinedMemberCount`
-                // value, therefore handling the logic locally here.
-                //
-                // Tracked as part of https://github.com/vector-im/element-web/issues/26033
-                const membersCount = includeFunctional
-                    ? room.getMembers().reduce((count, m) => {
-                          return m.membership === "join" ? count + 1 : count;
-                      }, 0)
-                    : getJoinedNonFunctionalMembers(room).length;
-                setCount(membersCount);
-            },
-            throttleWait,
-            { leading: true, trailing: true },
-        ),
+        throttle(onMembershipUpdate, throttleWait, { leading: true, trailing: true }),
+    );
+
+    useTypedEventEmitter(
+        room,
+        RoomEvent.Summary,
+        throttle(onMembershipUpdate, throttleWait, { leading: true, trailing: true }),
```

Comment:
Why wouldn't the `throttle` be on the `onMembershipUpdate` rather than each event having its own?